tools/cldr-apps/src/main/java/org/unicode/cldr/web/HttpStatusCache.java

```diff
@@ -0,0 +1,108 @@
+package org.unicode.cldr.web;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+
+/**
+ * This class caches the status (200, 404, etc.) of a set of pages, attempting to
+ * avoid unnecessary server traffic. At present the cache is a singleton.
+ * @author srl
+ *
+ */
+public class HttpStatusCache {
+    /**
+     * Main entrypoint. Check a URL for its status (i.e., is the page found or not?)
+     * @param url
+     * @return
+     */
+    public static final boolean check(final String url) {
+        try {
+            return isGoodStatus(check(new URL(url)));
+        } catch (MalformedURLException e) {
+            e.printStackTrace();
+            return false;
+        }
+    }
+
+    public static final Integer check(final URL url) {
+        try {
+            Integer status = urlCache.get(url, () -> internalCheck(url));
+            // TODO: if the status is a retry here, retry.
+            return status;
+        } catch (ExecutionException e) {
+            // TODO Auto-generated catch block
+            System.err.println("HttpStatusCache: Err in check of URL " + url.toString());
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    /**
+     * Flush the cache of a specific URL or all of them
+     * @param url ('all' if null)
+     */
+    public static final void flush(final URL url) {
+        if(url == null) {
+            System.err.println("HTTP Status Cache: invalidating all " + urlCache.size());
```

Comment:
println for debugging only?

Reply:
`flush()` is called from the JSP, so it is an administrative task.